src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -1050,5 +1050,123 @@ public static T WithCommentsFrom<T>(this T node, SyntaxToken leadingToken, Synta
+        /// and written to.
+        /// </summary>gi
+        public static bool IsReadFrom(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsReadFrom();
+
+        /// <summary>
+        /// True if a read is performed to the given expression.  Note: unlike <see cref="IsReadFrom(SyntaxNode, SemanticModel, CancellationToken)"/>, this
+        /// will not return true if writes are performed on the expression as well.  For example,
+        /// "++a" will return 'false'.  However, 'a' in "in a" or "x = a" will return true.
+        /// </summary>
+        public static bool IsOnlyReadFrom(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsOnlyReadFrom();
+
+        /// <summary>
+        /// True if a write is performed to the given expression.  Note: reads may also be performed
+        /// to the expression as well.  For example, "++a".  In this expression 'a' is both read from
+        /// and written to.
+        /// </summary>
+        public static bool IsWrittenTo(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsWrittenTo();
+
+        /// <summary>
+        /// True if a write is performed to the given expression.  Note: unlike <see cref="IsWrittenTo(SyntaxNode, SemanticModel, CancellationToken)"/>, this
+        /// will not return true if reads are performed on the expression as well.  For example,
+        /// "++a" will return 'false'.  However, 'a' in "out a" or "a = 1" will return true.
+        /// </summary>
+        public static bool IsOnlyWrittenTo(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsOnlyWrittenTo();
+
+        public static bool IsInOutContext(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsInOutContext();
+
+        public static bool IsInRefContext(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsInRefContext();
+
+        public static bool IsInRefOrOutContext(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsInRefOrOutContext();
+
+        public static bool IsInInContext(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+            => node.GetValueUsageInfo(semanticModel, cancellationToken).IsInInContext();
```

Comment:
i'm on the fence if these extensions have any value.  Since we have the helpers on ValueUsageInfo, all this is buying is not having to say GetValueUsageInfo.  However, i don't mind GetValueUsageInfo, it's like calling GEtSymbolInfo/GetTypeInfo/GetConstantValue/etc.

I feel like we don't need such a thin wrapper here.

Reply:
Oh, i see.  This wraps both getting the IOperation *and* getting the ValueUsageInfo.  Ideally, it feels like this would just be exposed off of hte SemanticModel.

Reply:
Yes, this thin wrapper makes it easier for clients to get this information without worrying about the underlying implementation details (IOperation/ValueUsageInfo).

Reply:
@mavasani Would it be worth considering exposing GetValueUsageInfo off of SemanticModel?  it would fit very cleanly in with the shape of SemanticModel and how people can use it, or IOperation, depending on which is most natural for their use case.

Reply:
Yes, that sounds like a better API. I will do this change when we decide to move this PR forward.

Reply:
SGTM.  Thanks for looking into this.  I like that this can help simplify the IDE, and help improve the overall compiler APIs!